vignettes/data-combined.Rmd

```diff
@@ -0,0 +1,300 @@
+
+```{r}
+# Check the units of the observed time values
+obsData$`Vergin 1995.Iv`$xUnit
+
+myDataCombinedTranformations$setDataTransformations(forNames = obsData$`Vergin 1995.Iv`$name,
+                                      xOffsets = 24)
+
+plotIndividualTimeProfile(dataCombined = myDataCombinedTranformations)
+```
+
+In the next step, we want to *normalize* observed concentrations to a dose. We 
+can easily achieve this with the scale factor. In the next example, we normalize 
+observed values to 250 mg dose by setting the `yScaleFactor` to 1/250:
+
+```{r}
+myDataCombinedTranformations$setDataTransformations(forNames = obsData$`Vergin 1995.Iv`$name,
+                                      yScaleFactors = 1/250)
+
+plotIndividualTimeProfile(dataCombined = myDataCombinedTranformations)
+```
+Finally, offsetting the observation values might be useful when working with measurements 
+of endogenous substrates, such as the hormone glucogen, and want to correct for the 
+individual specific baseline levels of the hormone.
+
+## Extracting a combined data frame
+
+The data frame (also sometimes called as a table) data structure is central to R-based workflows, and, thus, we may wish to extract a data frame for datasets present in the object.
+
+Internally, `DataCombined` extracts data frames for observed and simulated datasets and combines them.
+
+```{r}
+myDataCombined$toDataFrame()
+```
+
+This function returns a [tibble data frame](https://r4ds.had.co.nz/tibbles.html#tibbles-vs.-data.frame). If you wish to modify how it is printed, you can have a look at the available options [here](https://pillar.r-lib.org/reference/pillar_options.html). In fact, let's change a few options and print the data frame again.
+
+```{r, eval=FALSE}
+options(
+  pillar.width = Inf, # show all columns
+  pillar.min_chars = Inf # to turn off truncation of column titles
+)
+
+myDataCombined$toDataFrame()
+```
+
+
+```{r, echo=FALSE}
+# change these settings only temporarily
+withr::with_options(
+  list(
+    pillar.width = Inf, # show all columns
+    pillar.min_chars = Inf # to turn off truncation of column titles
+  ),
+  code = {
+    print(myDataCombined$toDataFrame())
+  }
+)
+```
+
+As can be seen, this data frame attemtps to provide a one-stop access to all important details about the datasets included in the object, including important meta-data.
+
+`{ospsuite}` also provides a few helper functions to modify the data frame further.
+
+When multiple (observed and/or simulated) datasets are present in a data frame, they are likely to have different units. `convertUnits()` function helps to convert them to a common unit^[Note that if you are using `DataCombined` object for plotting functions, you don't need to this conversion; the plotting functions will take care of this internally.].
+
+```{r}
+convertUnits(
+  myDataCombined,
+  xUnit = ospUnits$Time$s,
+  yUnit = ospUnits$`Concentration [mass]`$`µg/l`
+)
```

Comment:
looking at this: can I include only data of the same dimension in the **whole datacombined-object**?
Thought it would be per GRoup...

Reply:
When adding a data set AFTER convertUnits was called: will the units of the newly added dataset converted or will they be taken AS IS?

Reply:
> looking at this: can I include only data of the same dimension in the whole datacombined-object?
Thought it would be per GRoup...

Actually, it is possible to add AND group data with different dimensions.. Trying to convert units (and plotting) will fail. I think we should change this behavior and prevent from adding data sets with different dimensions. Will create an issue.

Reply:
> When adding a data set AFTER convertUnits was called: will the units of the newly added dataset converted or will they be taken AS IS?

`convertUnits()` does not modify the `DataCombined` object but returns a new data set. Added a note.

Reply:
https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1132